AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -106,3 +187,51 @@ def validate_variable_count(self, expected) -> None:
         result = helper.wait_for_condition(lambda: expected == row_count, WAIT_TIME_SEC_3)
 
         assert result, "Variable count did not match expected value."
+
+    def toggle_pin_variable_button(self, variable_type: str) -> None:
+        """
+        Function for clicking the pin toggle that appears for variables in the create variable list
+
+        param variable_type: The type of variable you want to pin to the top of the list
+
+        returns None
+        """
+        pinned_toggle_data = self.__find_pinned_toggle_control(variable_type)
+
+        assert pinned_toggle_data[1] is not None, "Failed to find variable pinning qt widget"
+
+        toggle_state_is_different = self.__toggle_the_pinned_control(pinned_toggle_data)
+
+        assert toggle_state_is_different, "Variable pin not successfully toggled"
+
+    def __find_pinned_toggle_control(self, variable_type: str):
+        """
+        helper function for toggling the pinned variable type in the create variable menu.
+
+        returns tuple with the variable view and toggle control in it
+        """
+        variable_table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
+        variable_object = pyside_utils.find_child_by_pattern(variable_table_view, variable_type)
+
+        toggle_index = 0
+        pinned_toggle = variable_object.siblingAtColumn(toggle_index)
+        pinned_toggle_data = {variable_table_view, pinned_toggle}
+        return pinned_toggle_data
+
+    def __toggle_the_pinned_control(self, pinned_toggle_data):
+        """
+        helper function for toggling the pinned variable type. performs a mouse click on the toggle control and waits
+        for the UI to respond
+
+        returns the boolean comparison of the state of the toggle control after it's been toggled.
+        """
+        pinned_toggle_state_before = pinned_toggle_data[1].data(Qt.DecorationRole)
+
+        # click the toggle then varify it's different than the before state
+        pyside_utils.item_view_index_mouse_click(pinned_toggle_data[0], pinned_toggle_data[1])
+
+        pinned_toggle_state_after = helper.wait_for_condition(lambda: pinned_toggle_data[1].data(Qt.DecorationRole) is None,
+                                                              WAIT_TIME_SEC_3)
+
+        return pinned_toggle_state_before != pinned_toggle_state_after
```

Comment:
To support using a tuple return, refactor this to support it.

```suggestion
    def __toggle_the_pinned_control(self, variable_table_view, pinned_toggle):
        """
        helper function for toggling the pinned variable type. performs a mouse click on the toggle control and waits
        for the UI to respond

        returns the boolean comparison of the state of the toggle control after it's been toggled.
        """
        pinned_toggle_state_before = pinned_toggle.data(Qt.DecorationRole)

        # click the toggle then varify it's different than the before state
        pyside_utils.item_view_index_mouse_click(variable_table_view, pinned_toggle)
        pinned_toggle_state_after = helper.wait_for_condition(lambda: pinned_toggle.data(Qt.DecorationRole) is None,
                                                              WAIT_TIME_SEC_3)

        return pinned_toggle_state_before != pinned_toggle_state_after
```

Reply:
implemented the tuple using this approach